File: marketdata-common/src/main/java/com/handson/infra/HttpRequest.java
package com.handson.infra;

import java.util.List;
import java.util.Map;

/**
 * Should grow if needed ;)
 */
public class HttpRequest {

    private final Map<String, List<String>> parameters;

    public HttpRequest(Map<String, List<String>> parameters) {
        this.parameters = parameters;
    }

    public String getParameter(String name) {
        if (parameters.get(name) == null) {
            return null;
        }
        return parameters.get(name).get(0);
    }
}

File: CSharp/NetCore/MarketData/src/MarketDataCommon/Infra/HttpRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace MarketDataCommon.Infra
{
    public class HttpRequest
    {
        private readonly Dictionary<string, List<string>> _parameters;

        public HttpRequest(Dictionary<string, List<string>> parameters)
        {
            _parameters = parameters;
        }

        public String GetParameter(string name)
        {
            if (!_parameters.ContainsKey(name))
                return null;

            return _parameters[name].First();
        }
    }
}
